### ISSUE TITLE
cover icon not changed with device class

### ISSUE THREAD

**Issue by yangtfu** [custom-cards/button-card#581]
**Checklist**
- [x] I updated the card to the latest version available
- [x] I cleared the cache of my browser

**Describe the bug**
When a cover entity is customized as curtain, its icon won't change in button card.

**Version of the card**
Version: 3.4.2

**To Reproduce**
This is the configuration I used:
```yaml
    - type: 'custom:button-card'
      entity: cover.curtain
      show_state: true
      show_icon: true
```

**Screenshots**
![image](https://user-images.githubusercontent.com/14918820/165016243-7242b7ae-49b0-4d4b-ab3e-68c438ae83bd.png)
It is actually a curtain
![image](https://user-images.githubusercontent.com/14918820/165016303-92084d30-51e4-46b9-931a-7d552103da17.png)

**Expected behavior**
icon shown as curtain
**Desktop (please complete the following information):**
 - Browser: Edge
 - Version: 100.0.1185.50

**Smartphone (please complete the following information):**
 - Device: oneplus 8t
 - OS: android 12
 - Browser: ha app
 - Version beta-2234

**Additional context**

**Comment by TTLucian** [comment]
Same problem with a switch:

![image](https://user-images.githubusercontent.com/1855276/181408608-ba2bd31d-60ac-4353-b210-2d1400268520.png)

![image](https://user-images.githubusercontent.com/1855276/181408679-792591dd-28e6-40c7-9d74-7e63792d7c82.png)

**Comment by sup3rm3ga** [comment]
I am also experiencing this for my cover which is set as a Gate.

**Comment by RomRider** [comment]
Fixed in the 4.0.0-dev or upcoming 4.0.0
